### DISCUSSION TITLE
Lesson 7: VRFConsumerBase & VRFConsumerBaseV2 & REMIX - out of gas exception

### DISCUSSION THREAD

**Discussion by rkovrigin** [smartcontractkit/full-blockchain-solidity-course-py#1696] [Q&A] [answered]
Guys, I have some GoerliETH on my account - 1.2338
I was able to deploy the contracts - VRFv2Consumer and RandomNumberConsumer via REMIX
But when I try to send some ETH to the contracts addresses I got - "Warning! Error encountered during contract execution [out of gas] ".
I checked I send less than I have on my account.
The transaction - https://goerli.etherscan.io/tx/0xd0505fe178861dc67e6b16cd0db5399a01405fdc0010a19f47153d8aa413c8b8
You can check that the address I send the ETH from has some gas

**Comment by Niferu** [comment, marked as answer]
Hello @rkovrigin 
 As I remember you have to first perform subscription (basically add your account to chainlink) then you have to fund it with some `LINK` instead of `ETH` and at last, you have to add deployed contract address to that subscription account you have added in first step, so this deployed contract can use `LINK` you funded to subscription account. Then it will be able to generate random number for you.

Also remember to pass correct `ID` from subscription, you can read it as showed below:
![image](https://user-images.githubusercontent.com/110537615/189333782-06f3c3b3-1931-4c43-9abc-c15224f18918.png)


You can also follow below video as it shows above steps in details:
https://www.youtube.com/watch?v=rdJ5d8j1RCg&ab_channel=Chainlink

**Reply by cromewar** [reply]
This is the correct answer, does this make sense for you @rkovrigin?

**Reply by rkovrigin** [reply]
Hello @Neftyr & @cromewar 
@Neftyr , I did what you have mentioned, indeed I didn't connect my Metamask with ChainLink, so I can see I have some balance. 
![Screenshot 2022-09-11 at 10 24 22](https://user-images.githubusercontent.com/3532458/189518462-b8986bf8-a34a-4cf4-84fa-31546827ca11.png)
But still have the same Error.

Do I have to do anything on a Metamask side, connect ChainLink account, or smth like that?

**Reply by rkovrigin** [reply]
Ooops, sorry.
So, I tried to fund my contract with GoerliETH, but it is incorrect.
I had to fund it with LINK, as @Neftyr mentioned. I just could not figure how to do so via MetaMask browser extension UI.
Apparently, first I had added the ChainLink tokens via Assets->Import Tokens, after that MetaMask UI let me do that.
Thanks @Neftyr !
